### DISCUSSION TITLE
Issue update LPS node firmware

### DISCUSSION THREAD

**Discussion by Kalib-Drift** [bitcraze/discussions#514] [Q&A] [answered]
Hey everyone

I am currently having an issue updating the LPS node firmware using the LPS configuration tool. I was previously unable to find any dfu files in the LPS-node-firmware project. After speaking with @tobbeanton he directed me to an updated version of the project which contains LPS-node-firmware-2022.09.dfu, after pulling this project I am now able to select the dfu file in the config tool but it will not let me update the node even in dfu mode. 

If anyone has any ideas or insight into this problem it would be much appreciated

**Comment by krichardsson** [comment]
This is usually related to USB on hour machine. Which OS are you using?

**Comment by tobbeanton** [comment]
A follow up question from the previous answer:
> You probably don't have permission to serial devices.
 https://askubuntu.com/questions/210177/serial-port-terminal-cannot-open-dev-ttys0-permission-denied

Did you add yourself to the plugdev group?

**Comment by Kalib-Drift** [comment, marked as answer]
Hey Krichardsson and Tobias.

I am currently running on ubuntu 22.04 

I have added myself to the plugdev group and no longer get an error when configuring the node id or mode. because I recently received the LPS pack, could it be the node already has all the latest firmware installed locally?

**Reply by krichardsson** [reply]
Great that the DFU problem seems to be solved! 

I'm a bit unclear on what the symptoms are at this stage. 

Are you following the instructions in the [tutorial](https://www.bitcraze.io/documentation/tutorials/getting-started-with-loco-positioning-system/)?
I suppose you are using the [LPS tool](https://github.com/bitcraze/lps-tools)?

It sounds like you can change settings but not flash the new firmware, correct?
Is the Loco node in DFU mode?

> could it be the node already has all the latest firmware installed locally

No , the boot loader is pretty stupid and will re-flash the same binary :-)

**Reply by Kalib-Drift** [reply]
Hey, krinchardsson, you are correct I am able to configure the LPS node but I am unable to flash the firmware. I have made sure that the device is in DFU mode but this did not seem to work. 

After reading your message it occurred to me that I should check to see if my computer is detecting the device at all when in DFU mode. after checking the tty list it seems as though my computer only sees the node when it is connected in normal mode but does not recognize it when in DFU mode, I'm not sure if this is the expected outcome but hopefully it helps.

**Reply by krichardsson** [reply]
You can use `dfu-util -l` to verify if DFU works.

With a Loco Node in DFU mode connected to my machine I get:
``` bash
kristoffer@kristoffer-XPS-13-9310:~/code$ dfu-util -l
dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Found DFU: [0483:df11] ver=2200, devnum=70, cfg=1, intf=0, path="3-8.1.5.4", alt=1, name="@Option Bytes  /0x1FFFF800/01*016 e", serial="FFFFFFFEFFFF"
Found DFU: [0483:df11] ver=2200, devnum=70, cfg=1, intf=0, path="3-8.1.5.4", alt=0, name="@Internal Flash  /0x08000000/064*0002Kg", serial="FFFFFFFEFFFF"
```

**Reply by Kalib-Drift** [reply]
AHA ! there is my issue, I did not have dfu-util properly installed, I am now able to flash the lps node. 

this doesn't seem to be important because I got the node to update, but when running dfu-util -l it claims I am using version 0.9 Copywrite 2010-2016.
